plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -625,8 +629,7 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         DiscretePredicates discretePredicates = null;
         if (!partitionSourceIds.isEmpty()) {
             // Extract identity partition columns
-            Map<Integer, IcebergColumnHandle> columns = getProjectedColumns(icebergTable.schema(), typeManager).stream()
-                    .filter(column -> partitionSourceIds.contains(column.getId()))
+            Map<Integer, IcebergColumnHandle> columns = getSelectedProjectColumns(icebergTable.schema(), typeManager, partitionSourceIds).stream()
```

Comment:
Current approach is fine, but I'm wondering why we're not able to pull out the column handle from `io.trino.plugin.iceberg.IcebergTableHandle#projectedColumns` ?

Reply:
It seems `io.trino.plugin.iceberg.IcebergTableHandle#projectedColumns` only keeps the columns which needs to be projected(`IcebergMetdata#applyProjection`). If partition column is not part of projection then we can not use `io.trino.plugin.iceberg.IcebergTableHandle#projectedColumns`.